modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java

```diff
@@ -793,6 +859,110 @@ public AndesMessageMetadata getMetadata(long messageId) throws AndesException {
      * {@inheritDoc}
      */
     @Override
+    public List<AndesMessageMetadata> getMetadataListForStorageQueueFromDLC(String storageQueueName,
+                                                                            String dlcQueueName, long firstMsgId,
+                                                                            long lastMsgId) throws AndesException {
+        List<AndesMessageMetadata> metadataList = new ArrayList<AndesMessageMetadata>();
+        Connection connection = null;
+        PreparedStatement preparedStatement = null;
+        ResultSet resultSet = null;
+
+        Context metaListRetrievalContext = MetricManager.timer(Level.INFO, MetricsConstants
+                .GET_META_DATA_LIST_FOR_QUEUE_FROM_DLC).start();
+        Context contextRead = MetricManager.timer(Level.INFO, MetricsConstants.DB_READ).start();
+
+        try {
+            connection = getConnection();
+            preparedStatement = connection
+                    .prepareStatement(RDBMSConstants.PS_SELECT_METADATA_RANGE_FROM_QUEUE_IN_DLC);
+            preparedStatement.setInt(1, getCachedQueueID(storageQueueName));
+            preparedStatement.setInt(2, getCachedQueueID(dlcQueueName));
+            preparedStatement.setLong(3, firstMsgId);
+            preparedStatement.setLong(4, lastMsgId);
+
+            resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                AndesMessageMetadata md = new AndesMessageMetadata(
+                        resultSet.getLong(RDBMSConstants.MESSAGE_ID),
+                        resultSet.getBytes(RDBMSConstants.METADATA),
+                        true
+                );
+                md.setStorageQueueName(storageQueueName);
+                metadataList.add(md);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("request: metadata range (" + firstMsgId + " , " + lastMsgId +
+                        ") in destination queue " + storageQueueName
+                        + "\nresponse: metadata count " + metadataList.size());
+            }
+        } catch (SQLException e) {
+            throw rdbmsStoreUtils.convertSQLException("Error occurred while retrieving messages between msg id " +
+                    firstMsgId + " and " + lastMsgId + "in DLC:" + dlcQueueName + " from queue " + storageQueueName, e);
+        } finally {
+            metaListRetrievalContext.stop();
+            contextRead.stop();
+            String task = RDBMSConstants.TASK_RETRIEVING_METADATA_RANGE_IN_DLC_FROM_QUEUE + storageQueueName;
+            close(resultSet, task);
```

Comment:
Introduced two new methods to for close(connection, prepared statement) and close(connection, prepared statement, result set) to avoid repetition. Addressed by the PR https://github.com/wso2/andes/pull/403